Unable to include optflow.hpp

Hi,

I want to compile example given in this post about [optflow](http://answers.opencv.org/question/148927/some-resolutions-cause-segmentation-faults-in-disopticalflow/).
But I have got problem compiling this program : 
```
#include "opencv2/core/utility.hpp"
#include "opencv2/highgui.hpp"
#include "opencv2/imgproc.hpp"
#include "opencv2/optflow.hpp"

using namespace std;
using namespace cv;
using namespace optflow; 

void main()
{
}

```


Error is 

>1>G:\Lib\opencv\modules\core\include\opencv2/core/hal/intrin_sse.hpp(1796): error C2065: CV_CPU_HAS_SUPPORT_SSE2' : identificateur non déclaré (undeclared identifier)

My cmake for this project is 

```
cmake_minimum_required(VERSION 2.8)

PROJECT (chSfm)

find_package(glog QUIET REQUIRED)
find_package(ceres QUIET REQUIRED)
find_package(OpenCV  REQUIRED)
find_package(VTK  OPTIONAL_COMPONENTS)


file(GLOB chSfm_SRCS
    "*.h"
    "*.cpp")
ADD_EXECUTABLE (chSfm ${chSfm_SRCS})


if (OpenCV_FOUND)
	include_directories( ${OpenCV_INCLUDE_DIRS} )
	include_directories( ${EIGEN_INCLUDE_DIRS} )
	if (VTK_FOUND)
		add_definitions(-DUSE_VTK)
		include_directories( ${VTK_INCLUDE_DIRS} )
	endif (Vtk_FOUND)
	if(ceres_FOUND)
	  add_definitions("-DCERES_FOUND=1")
	endif(ceres_FOUND)
	
	target_link_libraries( chSfm ${OpenCV_LIBS} )
	target_link_libraries( chSfm ${VTK_LIBRARIES} )
else (OpenCV_FOUND)
message("PB->OPENCV = ${OpenCV_INCLUDE_DIRS}")
endif (OpenCV_FOUND)

```



It is weird because I can compile and run example in opencv_contrib dis_opticalflow.cpp

My opencv cmake configuration is :

```
Could NOT find PythonInterp (missing:  PYTHON_EXECUTABLE) (Required is at least version "2.7")
Could NOT find PythonInterp (missing:  PYTHON_EXECUTABLE) (Required is at least version "2.6")
Could NOT find PythonInterp (missing:  PYTHON_EXECUTABLE) (Required is at least version "3.4")
Could NOT find PythonInterp (missing:  PYTHON_EXECUTABLE) (Required is at least version "3.2")
found Intel IPP (ICV version): 2017.0.2 [2017.0.2]
at: G:/Lib/opencv/dll2015/3rdparty/ippicv/ippicv_win
found Intel IPP IW binaries: 2017.0.2
at: G:/Lib/opencv/dll2015/3rdparty/ippicv/ippicv_win/../ippiw_win/
LAPACK(MKL): LAPACK_LIBRARIES: C:/Program Files (x86)/IntelSWTools/compilers_and_libraries/windows/mkl/lib/intel64/mkl_intel_lp64.lib;C:/Program Files (x86)/IntelSWTools/compilers_and_libraries/windows/mkl/lib/intel64/mkl_sequential.lib;C:/Program Files (x86)/IntelSWTools/compilers_and_libraries/windows/mkl/lib/intel64/mkl_core.lib
LAPACK(MKL): Support is enabled.
Could NOT find Matlab (missing:  MATLAB_MEX_SCRIPT MATLAB_INCLUDE_DIRS MATLAB_ROOT_DIR MATLAB_LIBRARIES MATLAB_LIBRARY_DIRS MATLAB_MEXEXT MATLAB_ARCH MATLAB_BIN) 
CMake Warning (dev) at G:/Lib/vtk/build/VTKConfig.cmake:39 (if):
  Policy CMP0054 is not set: Only interpret if() arguments as variables or
  keywords when unquoted.  Run "cmake --help-policy CMP0054" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  Quoted variables like "MSVC" will no longer be dereferenced when the policy
  is set to NEW.  Since the policy is not set the OLD behavior will be used.
Call Stack (most recent call first):
  cmake/OpenCVDetectVTK.cmake:6 (find_package)
  CMakeLists.txt:626 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

Found VTK ver. 7.1.0 (usefile: G:/Lib/vtk/CMake/UseVTK.cmake)
Caffe:   NO
Protobuf:   NO
Glog:   NO
Looking for tiny_dnn.h
Looking for tiny_dnn.h - found
Found tiny-dnn in: G:/Lib/opencv/dll2015/3rdparty/tinydnn/tiny-dnn-1.0.0a3
The protocol buffer compiler is not found (PROTOBUF_PROTOC_EXECUTABLE='Protobuf_PROTOC_EXECUTABLE-NOTFOUND')
freetype2:   NO
harfbuzz:    NO
No preference for use of exported gflags CMake configuration set, and no hints for include/library directories provided. Defaulting to preferring an installed/exported gflags CMake configuration if available.
Found installed version of gflags: C:/Program Files/gflags/CMake
Detected gflags version: 2.2.0
Found required Ceres dependency: Eigen version 3.3.0 in F:/lib/eigen-eigen
No preference for use of exported glog CMake configuration set, and no hints for include/library directories provided. Defaulting to preferring an installed/exported glog CMake configuration if available.
Found installed version of glog: F:/lib/glog
Detected glog version: 0.3.4
Found required Ceres dependency: Glog in 
Found Ceres version: 1.12.0 installed in: C:/Program Files/Ceres with components: [LAPACK, SchurSpecializations, OpenMP]
Checking SFM deps... TRUE
Torch importer has been enabled. To run the tests you have to install Torch ('th' executable should be available) and generate testdata using opencv_extra/testdata/dnn/generate_torch_models.py script.
Tesseract:   NO
No preference for use of exported gflags CMake configuration set, and no hints for include/library directories provided. Defaulting to preferring an installed/exported gflags CMake configuration if available.
Found installed version of gflags: C:/Program Files/gflags/CMake
Detected gflags version: 2.2.0
Found required Ceres dependency: Eigen version 3.3.0 in F:/lib/eigen-eigen
No preference for use of exported glog CMake configuration set, and no hints for include/library directories provided. Defaulting to preferring an installed/exported glog CMake configuration if available.
Found installed version of glog: F:/lib/glog
Detected glog version: 0.3.4
Found required Ceres dependency: Glog in 
Found Ceres version: 1.12.0 installed in: C:/Program Files/Ceres with components: [LAPACK, SchurSpecializations, OpenMP]
Checking SFM deps... TRUE

General configuration for OpenCV 3.2.0-dev =====================================
  Version control:               3.2.0-583-gfe4555e

  Extra modules:
    Location (extra):            G:/Lib/opencv_contrib/modules
    Version control (extra):     3.2.0-231-gdea8d5f

  Platform:
    Timestamp:                   2017-05-17T05:46:27Z
    Host:                        Windows 10.0.14393 AMD64
    CMake:                       3.7.1
    CMake generator:             Visual Studio 14 2015 Win64
    CMake build tool:            C:/Program Files (x86)/MSBuild/14.0/bin/MSBuild.exe
    MSVC:                        1900

  CPU/HW features:
    Baseline:                    SSE SSE2 SSE3 SSSE3
      requested:                 SSSE3
    Dispatched code generation:  SSE4_1 FP16 AVX AVX2
      requested:                 SSE4_1 AVX FP16 AVX2
      SSE4_1 (0 files):          + SSE4_1
      FP16 (0 files):            + SSE4_1 POPCNT SSE4_2 FP16 AVX
      AVX (1 files):             + SSE4_1 POPCNT SSE4_2 AVX
      AVX2 (1 files):            + SSE4_1 POPCNT SSE4_2 FP16 FMA3 AVX AVX2

  C/C++:
    Built as dynamic libs?:      YES
    C++ Compiler:                C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/x86_amd64/cl.exe  (ver 19.0.24215.1)
    C++ flags (Release):         /DWIN32 /D_WINDOWS /W4 /GR  /EHa  /D _CRT_SECURE_NO_DEPRECATE /D _CRT_NONSTDC_NO_DEPRECATE /D _SCL_SECURE_NO_WARNINGS /Gy /bigobj /Oi       /wd4251 /wd4324 /wd4275 /wd4589 /MP12  /MD /O2 /Ob2 /DNDEBUG  /Zi
    C++ flags (Debug):           /DWIN32 /D_WINDOWS /W4 /GR  /EHa  /D _CRT_SECURE_NO_DEPRECATE /D _CRT_NONSTDC_NO_DEPRECATE /D _SCL_SECURE_NO_WARNINGS /Gy /bigobj /Oi       /wd4251 /wd4324 /wd4275 /wd4589 /MP12  /D_DEBUG /MDd /Zi /Ob0 /Od /RTC1 
    C Compiler:                  C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/x86_amd64/cl.exe
    C flags (Release):           /DWIN32 /D_WINDOWS /W3  /D _CRT_SECURE_NO_DEPRECATE /D _CRT_NONSTDC_NO_DEPRECATE /D _SCL_SECURE_NO_WARNINGS /Gy /bigobj /Oi         /MP12  /MD /O2 /Ob2 /DNDEBUG  /Zi
    C flags (Debug):             /DWIN32 /D_WINDOWS /W3  /D _CRT_SECURE_NO_DEPRECATE /D _CRT_NONSTDC_NO_DEPRECATE /D _SCL_SECURE_NO_WARNINGS /Gy /bigobj /Oi         /MP12  /D_DEBUG /MDd /Zi /Ob0 /Od /RTC1 
    Linker flags (Release):      /machine:x64  /INCREMENTAL:NO  /debug
    Linker flags (Debug):        /machine:x64  /debug /INCREMENTAL 
    ccache:                      NO
    Precompiled headers:         YES
    Extra dependencies:          comctl32 gdi32 ole32 setupapi ws2_32 vfw32 C:/Program Files (x86)/IntelSWTools/compilers_and_libraries/windows/mkl/lib/intel64/mkl_intel_lp64.lib C:/Program Files (x86)/IntelSWTools/compilers_and_libraries/windows/mkl/lib/intel64/mkl_sequential.lib C:/Program Files (x86)/IntelSWTools/compilers_and_libraries/windows/mkl/lib/intel64/mkl_core.lib glog::glog gflags_shared
    3rdparty dependencies:

  OpenCV modules:
    To be built:                 core flann imgproc ml objdetect phase_unwrapping photo plot reg surface_matching video viz xphoto bgsegm dnn face fuzzy imgcodecs shape videoio xobjdetect highgui superres ts bioinspired dpm features2d line_descriptor saliency text calib3d ccalib datasets rgbd stereo structured_light tracking videostab xfeatures2d ximgproc aruco optflow sfm stitching
    Disabled:                    world contrib_world
    Disabled by dependency:      -
    Unavailable:                 cudaarithm cudabgsegm cudacodec cudafeatures2d cudafilters cudaimgproc cudalegacy cudaobjdetect cudaoptflow cudastereo cudawarping cudev java python2 python3 cnn_3dobj cvv dnn_modern freetype hdf matlab

  Non-free algorithms are enabled

  Windows RT support:            NO

  GUI: 
    QT:                          NO
    Win32 UI:                    YES
    OpenGL support:              NO
    VTK support:                 YES (ver 7.1.0)

  Media I/O: 
    ZLib:                        build (ver 1.2.8)
    JPEG:                        build (ver 90)
    WEBP:                        build (ver encoder: 0x020e)
    PNG:                         build (ver 1.6.24)
    TIFF:                        build (ver 42 - 4.0.2)
    JPEG 2000:                   build (ver 1.900.1)
    OpenEXR:                     build (ver 1.7.1)
    GDAL:                        NO
    GDCM:                        NO

  Video I/O:
    Video for Windows:           YES
    DC1394 1.x:                  NO
    DC1394 2.x:                  NO
    FFMPEG:                      YES (prebuilt binaries)
      avcodec:                   YES (ver )
      avformat:                  YES (ver )
      avutil:                    YES (ver )
      swscale:                   YES (ver )
      avresample:                YES (ver )
    GStreamer:                   NO
    OpenNI:                      NO
    OpenNI PrimeSensor Modules:  NO
    OpenNI2:                     NO
    PvAPI:                       NO
    GigEVisionSDK:               NO
    DirectShow:                  YES
    Media Foundation:            NO
    XIMEA:                       NO
    Intel PerC:                  NO

  Parallel framework:            Concurrency

  Other third-party libraries:
    Use Intel IPP:               2017.0.2 [2017.0.2]
               at:               G:/Lib/opencv/dll2015/3rdparty/ippicv/ippicv_win
    Use Intel IPP IW:            prebuilt binaries (2017.0.2)
    Use Intel IPP Async:         NO
    Use Lapack:                  YES (C:/Program Files (x86)/IntelSWTools/compilers_and_libraries/windows/mkl/lib/intel64/mkl_intel_lp64.lib C:/Program Files (x86)/IntelSWTools/compilers_and_libraries/windows/mkl/lib/intel64/mkl_sequential.lib C:/Program Files (x86)/IntelSWTools/compilers_and_libraries/windows/mkl/lib/intel64/mkl_core.lib)
    Use Eigen:                   YES (ver 3.3.0)
    Use Cuda:                    NO
    Use OpenCL:                  YES
    Use OpenVX:                  NO
    Use custom HAL:              NO

  OpenCL:                        <Dynamic loading of OpenCL library>
    Include path:                G:/Lib/opencv/3rdparty/include/opencl/1.2
    Use AMDFFT:                  NO
    Use AMDBLAS:                 NO

  Python 2:
    Interpreter:                 NO

  Python 3:
    Interpreter:                 NO

  Python (for build):            NO

  Java:
    ant:                         NO
    JNI:                         C:/Program Files/Java/jdk1.8.0_112/include C:/Program Files/Java/jdk1.8.0_112/include/win32 C:/Program Files/Java/jdk1.8.0_112/include
    Java wrappers:               NO
    Java tests:                  NO

  Matlab:                        Matlab not found or implicitly disabled

  Tests and samples:
    Tests:                       NO
    Performance tests:           NO
    C/C++ Examples:              YES

  Install path:                  G:/Lib/opencv/dll2015/install

  cvconfig.h is in:              G:/Lib/opencv/dll2015
-----------------------------------------------------------------

Configuring done
Generating done
```

just deleting the line from the header still leaves a warning:

`
C:\p\opencv_contrib\modules\optflow\src\sparse_matching_gpc.cpp:466:5: warning: "CV_SIMD128_64F" is not defined [-Wundef]
 #if CV_SIMD128_64F
`

so, moving the include to sparse_matching_gpc.cpp seems the better idea.

--- a/modules/optflow/include/opencv2/optflow/sparse_matching_gpc.hpp
+++ b/modules/optflow/include/opencv2/optflow/sparse_matching_gpc.hpp
@@ -52,7 +52,6 @@ the use of this software, even if advised of the possibility of such damage.
 #define __OPENCV_OPTFLOW_SPARSE_MATCHING_GPC_HPP__
 
 #include "opencv2/core.hpp"
-#include "opencv2/core/hal/intrin.hpp"
 #include "opencv2/imgproc.hpp"
 
 namespace cv

--- a/modules/optflow/src/dis_flow.cpp
+++ b/modules/optflow/src/dis_flow.cpp
@@ -40,7 +40,6 @@
 //
 //M*/
 
-#include "opencv2/core/hal/intrin.hpp"
 #include "precomp.hpp"
 #include "opencl_kernels_optflow.hpp"
 

--- a/modules/optflow/src/sparse_matching_gpc.cpp
+++ b/modules/optflow/src/sparse_matching_gpc.cpp
@@ -46,6 +46,7 @@
 #include "opencv2/imgcodecs.hpp"
 #include "precomp.hpp"
 #include "opencl_kernels_optflow.hpp"
+#include "opencv2/core/hal/intrin.hpp"
 
 /* Disable "from double to float" and "from size_t to int" warnings.
  * Fixing these would make the code look ugly by introducing explicit cast all around.

